### PR TITLE
Add optional `hooks` argument to `a11ySuite`.

### PR DESCRIPTION
**Pull request by bicknellr**

### CODE REVIEW

**Comment by azakus** [comment]
@justinfagnani

**Comment by bicknellr** [comment]
Thing to note: this no longer skips tests for which there are no matching elements. The test suites it generates will show all tests and the ones which don't apply will pass after doing nothing.

File: data/a11ySuite.js [hunk 1]
```diff
@@ -26,58 +36,73 @@
...
+          ignoredRules = ignoredRules || [];
+
+          var beforeEach = NOOP;
+          var afterEach = NOOP;
+          var fixtureElement;
+
+          if (!hooks) {
+            fixtureElement = document.getElementById(fixtureId);
+            if (!fixtureElement) {
+              console.warn('a11ySuite: No fixture with ID "' + fixtureId + '" exists.');
```
Comment (bicknellr) [inline data/a11ySuite.js]: I should probably pull these out..

File: data/a11ySuite.js [hunk 2]
```diff
@@ -26,57 +25,70 @@
...
+          var afterEach;
+
+          if (!hooks) {
+            var fixtureElement = document.getElementById(fixtureId);
+            if (!fixtureElement) {
+              return;
+            }
+
+            beforeEach = function() {
+              fixtureElement.create();
```
Comment (alice) [inline data/a11ySuite.js]: So it's the responsibility of the hooks to do this if you pass them in?

Comment (bicknellr) [inline data/a11ySuite.js]: Yeah, passing hooks skips the regular find-and-create-a-fixture process.

**Comment by rictic** [comment]
Please add tests of this new functionality (and possible of the old)

**Comment by FredKSchott** [comment]
Closing this PR as abandoned.

@bicknellr please take a look and rebase/reopen if this is something you'd still like to get merged in.

**Pull request closed without merging**